### ISSUE TITLE
`minishift (openshift) service foo` ignores the current namespace

### ISSUE THREAD

**Issue by jstrachan** [minishift/minishift#927]
when I'm in namespace `fabric8` I can see a service called `keycloak` but `minishift openshift service keycloak` can't find it.

```
$ oc project
Using project "fabric8" on server "https://192.168.64.82:8443".
$ oc get service keycloak
NAME       CLUSTER-IP      EXTERNAL-IP   PORT(S)   AGE
keycloak   172.30.88.128   <none>        80/TCP    12h
$ minishift openshift service keycloak --url
No service with name 'keycloak' defined in namespace 'default'
```
its confusing to use the namespace `default` rather than the current one - it leads to users having to type extra stuff really

**Comment by budhrg** [comment]
> $ minishift openshift service keycloak --url
No service with name 'keycloak' defined in namespace 'default'

@jstrachan , the command should be `minishift openshift service keycloak -n fabric8 --url`

One need to pass the namespace while using `minishift openshift service` command.

**Comment by jstrachan** [comment]
the need to pass in the current namespace as a parameter is exactly why I raised this issue - we shouldn't have to ;). it should default to use the current users namespace from the active context in $KUBECONFIG (or ~/.kube/config)

**Comment by jstrachan** [comment]
e.g. rather than typing this:
```
minishift openshift service keycloak -n fabric8 --url
```
I wanna type
```
minishift service keycloak -u
```
less typing FTW and convention over configuration (use the users default namespace not 'default')

**Comment by budhrg** [comment]
> less typing FTW and convention over configuration (use the users default namespace not 'default')

Got it @jstrachan  Definitely a better user experience IMO.

**Comment by jstrachan** [comment]
Also both `oc` and `kubectl` always use the users current namespace for operations. It'd be nice if `minishift` did the same to avoid surprise

**Comment by jorgemoralespou** [comment]
Agree with @jstrachan, but we still need to support passing ask the flags, just in case you want something from a different namespace.

**Comment by jstrachan** [comment]
Yeah, the '-n' option is very useful - it's just it should not be mandatory ;)

**Comment by budhrg** [comment]
> Yeah, the '-n' option is very useful - it's just it should not be mandatory ;)

Agree :+1: , `-n` is useful to explicitly get services from another project but not the current project.

**Comment by LalatenduMohanty** [comment]
@jstrachan @jorgemoralespou agree. We should be able to fix this.

**Comment by praveenkumar** [comment]
@hferentschik doesn't we already handled it as part of our NodePort adventure?

**Comment by hferentschik** [comment]
> doesn't we already handled it as part of our NodePort adventure?

It is.

**Comment by hferentschik** [comment]
Fixed by issue #1106
